src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(String operation, String fruit, int quantity) {
+        this.operation = Arrays.stream(Operation.values())
+                .filter(o -> o.getCode().equals(operation))
+                .findFirst().orElseThrow(NoSuchElementException::new);

```

Comment:
better move this mapping logic to the `Operation` `public static Operation getOption(String option)` method